### DISCUSSION TITLE
Obtaining an infeasible solution from heuristic for Solomon benchmark instance

### DISCUSSION THREAD

**Discussion by onecable5781** [google/or-tools#4178] [Routing (and legacy CP) questions] [answered]
Hello,

The Solomon benchmark instance solution for RC208, 25 customer problem is 269.1 (see http://web.cba.neu.edu/~msolomon/rc12solu.htm). The objective function is purely the transit time/cost only, and does not include waiting time/service time.

I programmed a C++ code that attempts to solve this problem using OR-opt heuristics. That is available here: https://godbolt.org/z/4sbqev84b

This file reads in the cost, time, demand, vehicle capacity, number of vehicles, opening time window, closing time window and service times from the attached input file:

[input.txt](https://github.com/google/or-tools/files/14949620/input.txt)

Note that in Solomon instances, the cost is actually the pure transit time. The transit time is Euclidean distances between points truncated to 1 decimal point. The input file provides 26 x 26 cost entries, 26 x 26 transit time entries, followed by 26 entries of time windows, service time and demand data. Barring transit time (and hence cost, as cost = transit time for these problems), all other problem parameters are integer.

Now, the optimal solution for the RC208, 25 customer problem is composed of 2 routes :

0 12 14 17 16 15 13 9 11 10 2 6 7 8 4 5 3 1 0
0 24 25 23 21 18 19 20 22 0

for a total cost (pure transit time) of 269.1

On running the above C++ code, I obtain the following infeasible route via ortools.

0 12 14 11 10 9 13 15 16 17 7 6 2 (t: 547.30) (late) 4 8 5 3 1 0 

That is, customer 2 is reached at time 547.3 while his closing time window is only 546. 

Note that the code will generate an output.txt file which contains the reaching time at each node along the route.

Could I request the developers to please take a look at the provided code to see if there is any obvious error I am committing? Could this be because some ortools input should only be an integer but I am providing double instead, or some type mismatch issue?

Thank you.

**Comment by lperron** [comment]
yes, the solver needs integer values.

**Reply by onecable5781** [reply]
Ah okay. For VRPTW solver, especially pertaining to time window related feasibility, I can understand that transit times need to be integer from your message. Is it needed that the objective function cost values need to be integer also or could they be double?

**Comment by lperron** [comment, marked as answer]
everything integral.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00



Le jeu. 11 avr. 2024 à 20:44, onecable5781 ***@***.***> a
écrit :

**Reply by onecable5781** [reply]
Thanks. I am able to confirm that on doing this, that in about 2 seconds, the heuristic does indeed find the optimal solution! Impressive.
